tests/Example/UploadHandler.php

```
@@ -0,0 +1,77 @@
+        private readonly string $uploadDirectory,
+        private readonly int $maxSize = 5 * 1024 * 1024, // 5MB
+        private readonly array $allowedTypes = ['image/jpeg', 'image/png', 'image/gif'],
+    ) {
+    }
+
+    /**
+     * @param array<string, mixed> $fileData $_FILES array element
+     */
+    public function handleUpload(array $fileData): UploadResult
+    {
+        $upload = FileUpload::create($fileData, [
```

PHPStan (2.1.31)

Parameter #2 $validationOptions of static method Koriym\FileUpload\FileUpload::create() expects array{maxSize?: int<1, max>, allowedTypes?: list<string>, allowedExtensions?: list<string>}, array{maxSize: int, allowedTypes: array} given.
• Offset 'maxSize' (int<1, max>) does not accept type int.
• Offset 'allowedTypes' (list<string>) does not accept type array<int|string, mixed>: array<int|string, mixed> might not be a list.

(argument.type)